✨ Feature: CLI 기반의 사보타지 게임 로직을 iOS 포팅

bisor0627 opened the pull request:
## About this PR

### ⚓ Related Issue
- #174 

### 🥥 Contents
- CLI 기반의 사보타지 게임 로직을 iOS용으로 이식하였습니다
- `Board`, `BoardCell`, `Card`, `Player` 등의 모델 구조를 설계하였고,
  SwiftUI 기반 `GameBoardView`로 시각화 및 조작 기능을 제공합니다
- 카드 배치 및 폭파 로직, 턴 전환, 승리 체크 등 주요 게임 흐름을 로컬에서 구현
- 현재는 연결 이후 P2P 정보를 활용하지 않고, Mock 유저 기반으로 동작합니다
- 게임 종료 후 다시 시작 또는 홈화면으로 이동할 수 있는 Alert 제공

### 🧭 Orientation 제한
- 다이나믹 아일랜드(좌측)에서 UI가 가려지는 문제를 방지하기 위해
  Info.plist에서 `LandscapeRight` 방향만 지원하도록 제한하였습니다
- 참고: https://990427.tistory.com/98

### 📸 Screenshot
<!-- UI 이미지가 있다면 첨부해주세요 -->

### 🚨 Checklist
- [x] 관련 이슈와 연결되었나요?
- [x] 빌드/테스트는 통과하였나요?
- [x] Mock 유저로 게임 흐름 테스트 완료
- [x] 방향 제한이 정상 적용되었나요?

### 💬 Other information
- 추후 P2P 연결 이후 실제 유저 데이터 연동 기능을 추가할 예정입니다

bisor0627 committed:
🐛 Fix. 다이나믹 아일랜드 기기에서 LandscapeRight만 허용

Why:
- Dynamic Island가 좌측에 있을 시, 가로모드에서 UI가 가려지는 이슈가 발생할 것을 고려함
- 사용자 경험을 해치지 않도록 LandscapeRight 방향만 지원하도록 제한

How:
- Info.plist에서 지원 방향을 LandscapeRight로 제한
- 사용하지 않는 NSLocalNetworkUsageDescription 키 제거

Tag: #ui #orientation #bugfix #info.plist #dynamicisland

bisor0627 committed:
✨ Feat. iOS용 사보타지 1차 임시 반영

Why:
- CLI 기반 사보타지 게임을 iOS로 옮겨 시각적 보드 기반 플레이 화면이 필요했음

How:
- Board, Card, Cell, Player 등 도메인 모델 정의 및 보드 UI 구현
- GameBoardView를 통해 카드 선택 및 배치 가능
- 현재 Mock 유저를 사용해 로컬 기기에서 플레이 테스트 가능
- 게임 종료 시 Alert로 다시 시작/홈 이동 제공

Tag: #ios #boardgame #swiftui #feature

JUDYLEE-cloud commented on Saboteur/Info.plist:
```diff
@@ -2,10 +2,6 @@
-    <key>NSLocalNetworkUsageDescription</key>
-        <string>This application will use local networking to discover nearby devices.</string>
-        
-        
```
이거 삭제하면 MPC 연결이 안될텐데 없애도 괜찮았나요?

bisor0627 replied:
헉 이부분 복구할게요!! 컨플릭트 웹에디터로 수정하다가 날아갔나봐요 ㅜㅜ

bisor0627 replied:
https://github.com/DeveloperAcademy-POSTECH/2025-C4-M4-Connectivity/pull/176/commits/b2b1c612baff024843562935da014f4b10c08f45
반영했습니다!

JUDYLEE-cloud commented on Saboteur/Features/GamePlay/GameView.swift:
```diff
@@ -16,27 +16,145 @@ struct GameView: View {
+                        Text("🎉 \(boardViewModel.currentPlayer.name)가 길을 완성했습니다!")
+                    }
             }
-            .onChange(of: winner.value) { _ in
-                gameState = .endGame
```
이 부분 없으면 제가 만든 게임 종료 화면이랑 다시 홈화면으로 이동하는 버튼을 볼 수 없어서 주석처리하고 올려줄 수 있을까요? 게임 종료 화면으로 이동하는 라우터 설정은 제가 수정할게요

bisor0627 replied:
네!! 복구해서 주석처리할게요!! 꼼꼼히 봐주셔서 감사해요!! 🥺💕💕

JUDYLEE-cloud commented on Saboteur/Features/GamePlay/GameView.swift:
```diff
@@ -16,27 +16,145 @@ struct GameView: View {
-                    winner.value = "플레이어 1"
-                    gameState = .endGame
-                } label: {
-                    Text("게임 종료 화면")
```
여기도! (주석처리하고 후에 제가 작업)

bisor0627 replied:
ZStack으로 `.topTrailing`에 뒀어요!!
https://github.com/DeveloperAcademy-POSTECH/2025-C4-M4-Connectivity/commit/79b4464c1f9eab2323a142ea97c12f44a897e86f

bisor0627 committed:
♻️ Refactor. NSLocalNetworkUsageDescription 키 복원하여 네트워크 권한 명시

Why:
- 코드 리뷰에서 `NSLocalNetworkUsageDescription` 누락에 대한 피드백을 받음
- Bonjour 서비스를 사용하는 경우, 해당 키가 없으면 iOS에서 네트워크 접근을 차단할 수 있음

How:
- Info.plist에 NSLocalNetworkUsageDescription 키 및 안내 문구 추가

Tag: #refactor #plist #network #localnetwork #p2p

Co-authored-by: JUDY LEE <ljh.social@ewhain.net>

See: https://github.com/DeveloperAcademy-POSTECH/2025-C4-M4-Connectivity/pull/176#discussion_r2206694234

bisor0627 committed:
♻️ Refactor. 게임 종료 버튼 UI 복원

Why:
- 리뷰에서 삭제된 "게임 종료 화면" 버튼의 테스트 용도 및 위치에 대한 복원이 요청됨
- 개발 중 수동 테스트를 위한 진입 지점을 유지할 필요 있음

How:
- `GameView` 내에 ZStack 상단에 버튼을 배치하고
  기존 GameBoardView와 함께 표시되도록 수정
- 버튼 클릭 시 강제로 winner를 설정하고 `.endGame` 상태로 전환

Tag: #refactor #ui #gameplay #codereview
See: https://github.com/DeveloperAcademy-POSTECH/2025-C4-M4-Connectivity/pull/176#discussion_r2206716977

JUDYLEE-cloud reviewed (approved):
머지 가자~~(˘▾˘~)(~˘▾˘)~
